src/frontend/src/expr/table_function.rs

```diff
@@ -62,6 +65,113 @@ impl TableFunction {
+                    Some(Ok(value)) => {
+                        if value.is_none() {
+                            return Err(BindError(
+                                "file_scan function does not accept null arguments".to_string(),
+                            )
+                            .into());
+                        }
+                        match value {
+                            Some(ScalarImpl::Utf8(s)) => {
+                                eval_args.push(s.to_string());
+                            }
+                            _ => {
+                                return Err(BindError(
+                                    "file_scan function only accepts string arguments".to_string(),
+                                )
+                                .into())
+                            }
+                        }
+                    }
+                    Some(Err(err)) => {
+                        return Err(err);
+                    }
+                    None => {
+                        return Err(BindError(
+                            "file_scan function only accepts constant arguments".to_string(),
+                        )
+                        .into());
+                    }
+                }
+            }
+            if !"parquet".eq_ignore_ascii_case(&eval_args[0]) {
+                return Err(BindError(
+                    "file_scan function only accepts 'parquet' as file format".to_string(),
+                )
+                .into());
+            }
+
+            if !"s3".eq_ignore_ascii_case(&eval_args[1]) {
+                return Err(BindError(
+                    "file_scan function only accepts 's3' as storage type".to_string(),
+                )
+                .into());
+            }
+
+            #[cfg(madsim)]
+            return Err(crate::error::ErrorCode::BindError(
+                "file_scan can't be used in the madsim mode".to_string(),
+            )
+            .into());
+
+            #[cfg(not(madsim))]
+            tokio::task::block_in_place(|| {
+                tokio::runtime::Handle::current().block_on(async {
+                    let parquet_stream_builder = create_parquet_stream_builder(
+                        eval_args[2].clone(),
+                        eval_args[3].clone(),
+                        eval_args[4].clone(),
+                        eval_args[5].clone(),
+                    )
+                    .await?;
+
+                    let mut rw_types = vec![];
+                    for field in parquet_stream_builder.schema().fields() {
+                        rw_types.push((
+                            field.name().to_string(),
+                            IcebergArrowConvert.type_from_field(field)?,
+                        ));
+                    }
+
+                    Ok::<risingwave_common::types::DataType, anyhow::Error>(DataType::Struct(
+                        StructType::new(rw_types),
+                    ))
```

Comment:
Is it ok to suppose that all the data in the direction has the same schema?

Reply:
A better way is to allow users to specify a read schema. Clickhouse doc: https://clickhouse.com/docs/en/sql-reference/table-functions/s3